### PR TITLE
Expression testing and clean ups

### PR DESCRIPTION
**Pull request by harshangrjn**
1. JuMP v0.21 build error issues fixed
2. #146 (MOI support fix)
3. expression.jl clean up 
4. eval() removal where unnecessary #153 
5. NL objective parsing fix for higher-order monomials: #104 
6. Issues #151 #150 addressed
7. Drop support for all trigonometric related functions (both in src and testing) 
8. Drop support of expression parsing for generic integer variables in MINLPs (both in src and testing)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,10 +4,10 @@ os:
   - linux
   - osx
 julia:
-  - 1.0
+  - 1.5
```
Comment (blegat) [inline .travis.yml:7]: 1.5 and 1 are the same thing, 1 links to the latest 1.x release

File: test/runtests.jl [hunk 2]
```diff
@@ -37,5 +36,5 @@ end
 # Perform Tests
 include("$(alpine_dir)/test/solver.jl")
 include("$(alpine_dir)/test/expression.jl")
-include("$(alpine_dir)/test/algorithm.jl")
-include("$(alpine_dir)/test/utility.jl")
+# include("$(alpine_dir)/test/algorithm.jl")
```
Comment (blegat) [inline test/runtests.jl:39]: Why are these commented ? This is the cause of the coverage drop

Comment (harshangrjn) [reply test/runtests.jl:39]: @blegat Good point! I am just cleaning up tests individually. I just created the PR but still pushing updates to the PR. It should be uncommented in the newer commits.

File: .travis.yml [hunk 3]
```diff
@@ -5,9 +5,9 @@ os:
   - osx
 julia:
   - 1.0
-  - 1
+  - 1 # Most recent v > 1.0
 before_script:
- - julia -e 'using Pkg; Pkg.add(PackageSpec(name="Pavito", rev="master"))' # TODO remove when Pavito#master is released.
+ - julia -e 'using Pkg; Pkg.add(PackageSpec(name="Pavito", rev="master"))' # TODO remove when Pavito #master is released.
```
Comment (blegat) [inline .travis.yml:10]: You can remove this, Pavito has been release

File: .travis.yml [hunk 4]
```diff
@@ -16,7 +16,7 @@ jobs:
       julia: 1
       os: linux
       script:
-        - julia -e 'using Pkg; Pkg.rm("Pavito")' # TODO Pkg seems to get into troubles because of the `before_script`. Remove when Pavito#master is released.
+        - julia -e 'using Pkg; Pkg.rm("Pavito")' # TODO Pkg seems to get into troubles because of the `before_script`. Remove when Pavito #master is released.
```
Comment (blegat) [inline .travis.yml:19]: You can remove this too

Comment (harshangrjn) [reply .travis.yml]: Done!

File: .travis.yml (hunk 3 above)
Comment (harshangrjn) [reply .travis.yml]: Done!

**Pull request merged**